frontend/src/components/pages/PrototypePage.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useRef, useReducer } from "react";
+import GridLayout from "react-grid-layout";
+import "react-grid-layout/css/styles.css";
+import "react-resizable/css/styles.css";
+import DraggableSource from "../common/grid/DraggableSource";
+import GridElement from "../common/grid/GridElement";
+import layoutReducer from "../common/grid/layoutReducer";
+
+interface LayoutItem extends GridLayout.Layout {
+  content: string;
+  temp?: boolean;
+  mouseEvent?: { clientX: number; clientY: number };
+}
+
+const Grid = () => {
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [layout, dispatch] = useReducer(layoutReducer, []);
+
+  const gridContainerStyle = {
+    backgroundColor: "lightgray",
+    borderLeft: "1px solid black",
+    borderRight: "1px solid black",
+    borderBottom: "1px solid black",
+    width: "601px",
+    height: "601px",
+    backgroundImage:
+      "linear-gradient(to right, black 1px, transparent 1px), linear-gradient(to bottom, black 1px, transparent 1px)",
+    backgroundSize: "50px 50px",
+  };
+
+  return (
+    <div
+      style={{
+        width: "full",
+        height: "full",
+        justifyContent: "center",
+        alignItems: "center",
+      }}
+    >
+      <DraggableSource
+        targetRef={ref}
+        dispatch={dispatch}
+        key="1"
+        componentType="TextBox"
+      >
+        <div
+          style={{
+            width: "100px",
+            height: "100px",
+            border: "1px solid blue",
+            alignItems: "center",
+            justifyContent: "center",
+          }}
+        >
+          {" "}
+          Text Box - Drag Me !
+        </div>
+      </DraggableSource>
+      <DraggableSource
+        targetRef={ref}
+        dispatch={dispatch}
+        key="2"
+        componentType="Match"
+      >
+        <div
+          style={{
+            width: "100px",
+            height: "100px",
+            border: "1px solid blue",
+            alignItems: "center",
+            justifyContent: "center",
+          }}
+        >
+          {" "}
+          Match - Drag Me !
+        </div>
+      </DraggableSource>
+      <div ref={ref}>
+        <GridLayout
+          className="layout"
+          style={gridContainerStyle}
+          layout={layout}
+          onLayoutChange={(layout) => dispatch({ type: "newLayout", layout })}
+          cols={12}
+          rowHeight={50}
+          maxRows={12}
+          verticalCompact={false}
```

Comment:
They might be deprecating this attribute in a future version. I think `compactType={null]` is the way they want us to do it now.

![image](https://github.com/uwblueprint/extend-a-family/assets/70064220/4bd55919-f745-4514-aecb-a26e0f640759)